scipy/stats/tests/test_distributions.py

```diff
@@ -3990,6 +3991,82 @@ def test_burr_nan_mean_var_9544(self):
+    vs = [1, 3, 10, 20]
+    alpha05 = [17.97, 45.40, 54.33, 59.56,
+               4.501, 8.853, 10.35, 11.24,
+               3.151, 5.305, 6.028, 6.467,
+               2.950, 4.768, 5.357, 5.714]
+    alpha01 = [90.03, 227.2, 271.8, 298.0,
+               8.261, 15.64, 18.22, 19.77,
+               4.482, 6.875, 7.712, 8.226,
+               4.024, 5.839, 6.450, 6.823]
+    ks_d = np.ravel(list((repeat(ks, 4))))
+    vs_d = np.ravel(list(zip(*(repeat(vs, 4)))))
+    p95 = [.95] * 16
+    p99 = [.99] * 16
+    sig_k_v_q = (list(zip(p95, ks_d, vs_d, alpha05))
+                 + list(zip(p99, ks_d, vs_d, alpha01)))
+
+    @pytest.mark.parametrize("significance, k, v,q", sig_k_v_q)
+    def test_cdf_against_tables(self, significance, k, v, q):
+        res_p = stats.studentized_range.cdf(q, k, v)
+        # some table values are only accurate to the second decimal place.
+        atol = 1e-3 if q < 10 else 1e-2 if q < 100 else 1e-1
+        assert_allclose(res_p, significance, atol=atol)
+
+    @pytest.mark.parametrize("significance,k,v,q", sig_k_v_q)
+    def test_ppf_against_tables(self, significance, k, v, q):
+        res_q = stats.studentized_range.ppf(significance, k, v)
+        # some table values are only accurate to the second decimal place.
+        atol = 1e-3 if q < 10 else 1e-2 if q < 100 else 1e-1
+        assert_allclose(res_q, q, atol=atol)
+
+    def test_pdf_integration(self):
+        k, v = 3, 10
+
+        # Test whether PDF integration is 1 like it should be.
+        res = quad(stats.studentized_range.pdf, 0, np.inf, args=(k, v))
+        assert_allclose(res[0], 1)
+
+    @pytest.mark.slow
+    def test_pdf_against_cdf(self):
+        k, v = 3, 10
+
+        # Test whether the integrated PDF matches the CDF using cumulative
+        # integration. Use a small step size to reduce error due to the
+        # summation. This is slow, but tests the results well.
+        x = np.arange(0, 10, step=0.01)
+
+        y_cdf = stats.distributions.studentized_range.cdf(x, k, v)[1:]
+        y_pdf_raw = stats.distributions.studentized_range.pdf(x, k, v)
+        y_pdf_cumulative = cumulative_trapezoid(y_pdf_raw, x)
+
+        # Because of error caused by the summation, use a relatively large rtol
+        assert_allclose(y_pdf_cumulative, y_cdf, rtol=1e-4)
+
+    @pytest.mark.slow
+    def test_moment_against_pdf(self):
+        k, nu = 3, 10
+        moments = [1, 2, 3, 4]
+
+        for moment in moments:
+            res_act = stats.studentized_range.moment(moment, k, nu)
+
+            def wrapper(x):
+                return x ** moment * stats.studentized_range.pdf(x, k, nu)
+            res_exp = quad(wrapper, 0, np.inf)[0]
+
+            # Atol is large B/C of integration innacuracy with quad.
+            # Setting epsabs to 1e-15 of the PDF + moment quads in studentized
+            # range causes the results to converge and pass default atol.
+            assert_allclose(res_act, res_exp, atol=1e-5)
+
```

Comment:
Let's test the entropy against trapezoidal rule.

Actually, another thing you could do is generate a data file with a handful of really accurate results with mpmath and test against those values. I think that would strengthen a lot of the tests here because many of them have very loose tolerances. You would also commit the file used to generate the data file.  An example of this sort of thing (using mpmath to generate data, even though it's not used for testing) is in gh-10170.